scripts/create_changelog.py

```diff
@@ -0,0 +1,343 @@
+        original_content = f.read()
+    tree = ET.fromstring(original_content)
+
+    for new_name, github in new_contributors:
+        new_contributor = ET.Element('contributor')
+        new_contributor.set('name', new_name)
+        new_contributor.set('github', github)
+
+        tree.append(new_contributor)
+    
+    sorted_contributors = sorted(
+        tree.findall('contributor'),
+        # Items which don't have a name attribute are placed at the end 
+        key=lambda x: unidecode(get_last_name(x.get('name', '\x7F'))) # '\x7F' is highest ASCII value
+    )
+
+    for cont in tree.findall('contributor'):
+        tree.remove(cont)
+
+    for cont in sorted_contributors:
+        tree.append(cont)
+
+    xml_lines = [
+        '<?xml version="1.0" encoding="UTF-8" ?>',
+        '<!--',
+        'Sage contributors',
+        '',
+        '+ mandatory / - optional',
+        '',
+        'contributor',
+        ' + name = full name',
+        ' - altnames = comma-separated list of alternate full names',
+        ' - location = something which returns the correct location when entered in maps.google.com',
+        ' - work = university, institute, ...',
+        ' - description = <item>[;<item>]* ... delimiter ";" is converted into a break or list',
+        ' - url = webpage, when clicking on name; no need to provide github profile URL',
+        ' - pix = url to a small image (max 50x50 or something like that) .. subdirectory "./pix/people/" should store them locally',
+        ' - jitter = float as string, multiplicated in jitter function, 0 to disable',
+        ' - trac = trac nickname for the search (also searched for the name, but the trac name is often different!). comma-separated list of several trac account names ok. No need to provide "gh-..." names, provide a github attribute instead.',
+        ' - github = github.com account name',
+        ' - gitlab = gitlab.com account name',
+        '',
+        'Please keep the list in alphabetical order by last name!',
+        '-->',
+        '<contributors>'
+    ]
+
+    # Write contributors with precise formatting
+    for contributor in sorted_contributors:
+        contrib_line = '<contributor'
+        for attr, value in contributor.attrib.items():
+            contrib_line += f'\n {attr}="{html.escape(value,quote=False)}"'
+        contrib_line += '/>'
+        xml_lines.append(contrib_line)
+    xml_lines.append('</contributors>')
+    
+    with open(original_file, 'w',encoding='utf-8') as f:
+        f.write('\n'.join(xml_lines))
+
+
+def map_git_to_names():
+    tree = ET.parse('conf/contributors.xml')
+    root = tree.getroot()
+    for c in root.findall('contributor'):
+        name = c.get('name')
+        git = c.get('github')
+        if git and name:
+            git_to_name[git] = unidecode(name)
+
+
+def fetch_real_name(github_name):
+    print(github_name)
+    url = f"https://api.github.com/users/{github_name}"
```

Comment:
this print is for debugging?

Reply:
Yes it was for debugging, will remove it